### PR TITLE
DT: Ensure proper quoting in grouping coalesce

### PR DESCRIPTION
**Pull request by iyerr3**
JIRA: MADLIB-1217

Grouping column value is coalesced with null_proxy to get the right null
identifier when null_as_category is True. The null_proxy needs to be
quoted appropriately as a string.

Closes #248

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/398/

**Comment by iyerr3** [comment]
I've combined two independent commits into a single PR, since the changes in the commits are close by and would benefit from the same reviewer looking at both changes.

**Comment by iyerr3** [comment]
Commits 97064e2 and 2ad0bf7 will eventually be combined in ffd6355 before merging.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/399/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/400/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/401/

**Comment by fmcquillan99** [comment]
I checked against the examples in 
JIRA: MADLIB-1217
JIRA: MADLIB-1218
and both work OK for me.  

So from the fix to the functionality perspective, LGTM.  

Other reviewers can comment on the software implementation, thx.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/407/

File: src/ports/postgres/modules/recursive_partitioning/decision_tree.py_in [hunk 1]
```diff
@@ -970,16 +970,35 @@ def _get_bins_grps(
...
+            for grp_key, items in groupby(all_levels, key=itemgetter('grp_key'))]
     if cat_features:
-        cat_items_list = [rows[col] for col in cat_features
-                          for grp_key, rows in grp_to_col_to_row.items() if col in rows]
+        # Below statements collect the grp_to_col_to_levels into multiple variables
+        # From above eg.
+        #   cat_items_list = [[0,1], [4,6,8], [0,1], [4,6], [0,1], [4,6,8]]
+        #   cat_n = [2, 3, 2, 2, 2, 3]
+        #   cat_n = [0, 1, 4, 6, 8, 0, 1, 4, 6, 0, 1, 4, 6, 8]
+        #   grp_key_cat = ['3', '4', '5']
```
Comment (njayaram2) [inline src/ports/postgres/modules/recursive_partitioning/decision_tree.py_in]: +1 for the examples, very helpful.

File: src/ports/postgres/modules/recursive_partitioning/decision_tree.py_in [hunk 2]
```diff
@@ -970,16 +970,35 @@ def _get_bins_grps(
...
+            (grp_key, dict((row['colname'], row['levels']) for row in items))
+            for grp_key, items in groupby(all_levels, key=itemgetter('grp_key'))]
     if cat_features:
-        cat_items_list = [rows[col] for col in cat_features
-                          for grp_key, rows in grp_to_col_to_row.items() if col in rows]
+        # Below statements collect the grp_to_col_to_levels into multiple variables
+        # From above eg.
+        #   cat_items_list = [[0,1], [4,6,8], [0,1], [4,6], [0,1], [4,6,8]]
+        #   cat_n = [2, 3, 2, 2, 2, 3]
+        #   cat_n = [0, 1, 4, 6, 8, 0, 1, 4, 6, 0, 1, 4, 6, 8]
```
Comment (njayaram2) [inline src/ports/postgres/modules/recursive_partitioning/decision_tree.py_in]: Should this be `cat_origin` instead?

**Pull request closed without merging**